docs/feature-roles.asciidoc

```diff
@@ -0,0 +1,444 @@
+|Write events into {es}
+
+|Index
+|`create_index` on +{beat_default_index_prefix}-*+ indices
+|Create APM indices in {es}
+|====
+
+. If <<configuration-rum,real user monitoring>> and <<sourcemaps,sourcemaps>> are enabled,
+assign the following additional privileges to the *general writer role*:
++
+[options="header"]
+|====
+|Type | Privilege | Purpose
+
+|Index
+|`read` on +{beat_default_index_prefix}-*sourcemap+ indices
+|Read sourcemaps from {es}
+|====
+
+. Assign the *general writer role* to users who need to publish {beatname_uc} data.
+
+[float]
+==== Specific writer roles
+
+Instead of creating a *general writer role*,
+individual publishing tasks, like writing events or uploading sourcemaps,
+can be performed by dedicated users with stricter privileges.
+
+[float]
+===== Sourcemap writer role
+
+To create an {beatname_uc} user that can write sourcemaps to {es}:
+
+. Create a *sourcemap writer role*, called something like +{beat_default_index_prefix}_sourcemap+,
+that has the following privileges:
++
+[options="header"]
+|====
+|Type | Privilege | Purpose
+
+|Index
+|`create_doc` on +{beat_default_index_prefix}-*+ indices
+|Write APM events into {es}
+
+|Index
+|`create_index` on +{beat_default_index_prefix}-*+ indices
+|Create APM indices in {es}
+|====
+
+. Assign the *sourcemap writer role* to users who need to publish sourcemaps.
+
+////
+***********************************  ***********************************
+***********************************  ***********************************
+////
+
+[[privileges-to-publish-monitoring]]
+=== Grant privileges and roles needed for monitoring
+
+++++
+<titleabbrev>Create a _monitoring_ user</titleabbrev>
+++++
+
+{es-security-features} provides built-in users and roles for publishing and viewing monitoring data.
+The privileges and roles needed to publish monitoring data
+depend on the method used to collect that data.
+
+* <<privileges-to-publish-monitoring-write>>
+** <<privileges-to-publish-monitoring-internal>>
+** <<privileges-to-publish-monitoring-metricbeat>>
+* <<privileges-to-publish-monitoring-view>>
+
+[float]
+[[privileges-to-publish-monitoring-write]]
+==== Publish monitoring data
+
+[IMPORTANT]
+====
+**{ecloud} users:** This section does not apply to our
+https://www.elastic.co/cloud/elasticsearch-service[hosted {ess}].
+Monitoring on {ecloud} is enabled by clicking the *Enable* button in the *Monitoring* panel.
+====
+
+[float]
+[[privileges-to-publish-monitoring-internal]]
+===== Internal collection
+
+If you're using <<monitoring-internal-collection,internal collection>> to
+collect metrics about {beatname_uc}, {security-features} provides
+the +{beat_monitoring_user}+ {ref}/built-in-users.html[built-in user] and
++{beat_monitoring_user}+ {ref}/built-in-roles.html[built-in role] to send
+monitoring information. You can use the built-in user, if it's available in your
+environment, or create a user who has the privileges needed to send monitoring
+information.
```

Comment:
minor note: you don't touch on the `beat_monitoring_user` role any more. Should we change to something like:

> You can use the built-in user, if it's available in your
environment, or create a user which has the built-in role assigned, 
or create a user and manually assign the required privileges.
information.